app/javascript/test/SpellingSuggestion.test.tsx

```diff
@@ -1,21 +1,254 @@
+/* eslint-disable camelcase */
 import '@testing-library/jest-dom';
-import { render } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 import React from 'react';
+import { I18n } from 'i18n-js';
+import { LanguageContext } from '../contexts/LanguageContext';
 import { SpellingSuggestion } from '../components/Results/SpellingSuggestion/SpellingSuggestion';
+import { enableFetchMocks } from 'jest-fetch-mock';
+enableFetchMocks();
+
+const locale = {
+  en: {
+    showingResultsFor: 'Showing results for %{corrected_query}',
+    searchInsteadFor: 'Search instead for %{original_query}'
+  }
+};
+
+const i18n = new I18n(locale);
 
 jest.mock('i18n-js', () => {
   return jest.requireActual('i18n-js/dist/require/index');
 });
 
-describe('Alert component', () => {
+describe('SpellingSuggestion component', () => {
+  const headers = {
+    Accept: 'application/json',
+    'Content-Type': 'application/json'
+  };
+  global.fetch = jest.fn(() =>
+    Promise.resolve({
+      json: () => Promise.resolve({})
+    })
+  ) as jest.Mock;
+
   const spellingSuggestionProps = {
-    suggested: '<a href="/search?affiliate=test_alert&amp;query=government">government</a>',
-    original: '<a href="/search?affiliate=test_alert&amp;query=%2Bgovermment">govermment</a>'
+    suggested: '<a class="suggestedQuery" href="/search?affiliate=test_affiliate&amp;query=medical">medical</a>',
+    original: '<a class="originalQuery" href="/search?affiliate=test_affiliate&amp;query=mecidal">mecidal</a>',
+    originalUrl: '/search?affiliate=test_affiliate&query=mecidal',
+    originalQuery: 'mecidal',
+    suggestedQuery: 'medical',
+    suggestedUrl: '/search?affiliate=test_affiliate&query=medical',
+    affiliate: 'test_affiliate'
   };
 
   it('renders SpellingSuggestion component', () => {
     render(
-      <SpellingSuggestion {...spellingSuggestionProps} />
+      <LanguageContext.Provider value={i18n} >
+        <SpellingSuggestion {...spellingSuggestionProps} vertical='web' />
+      </LanguageContext.Provider>
+    );
+  });
+
+  it('Show original and suggested query', () => {
+    render(
+      <LanguageContext.Provider value={i18n} >
+        <SpellingSuggestion {...spellingSuggestionProps} vertical='web'/>
+      </LanguageContext.Provider>
+    );
+
+    expect(screen.getByText('Showing results for')).toBeInTheDocument();
+    expect(screen.getByText('medical')).toBeInTheDocument();
```

Comment:
Can we merge this expectation with the one on top?
```
expect(screen.getByText('Showing results for medical')).toBeInTheDocument();
```

Reply:
Let me check this.

Reply:
Actually, "Showing results for medical" is not a string, "medical" is a hyperlink whose links I am checking already.

Reply:
Let me check to put it as one string with HTML to test it.

Reply:
This is updated to check the full string.